### PR TITLE
glfs: add support for resize/reconfig of block backend

### PR DESCRIPTION
**Pull request by pkalever**
$ targetcli /backstores/user:glfs/ ls
o- user:glfs ........................... [Storage Objects: 1]
  o- test  [sample@192.168.0.105/test (1.0GiB) activated]

$ targetcli /backstores/user:glfs/test get attribute dev_size
dev_size=1073741824

$ truncate -s +1G /mnt/test

$ ls -l /mnt/test
-rw-------. 1 root root 2147483648 Nov  2 15:05 /mnt/test

$ targetcli /backstores/user:glfs/test set attribute dev_size=2147483648
Parameter dev_size is now '2147483648'.

Signed-off-by: Prasanna Kumar Kalever <prasanna.kalever@redhat.com>

### CODE REVIEW

**Comment by pkalever** [comment]
Thanks to https://github.com/open-iscsi/tcmu-runner/pull/309 for making life of this patch much simpler.

File: glfs.c [hunk 1]
```diff
@@ -634,6 +635,41 @@ static int tcmu_glfs_write(struct tcmu_device *dev,
...
+	int ret;
+
+	ret = glfs_lstat(gfsp->fs, gfsp->hosts->path, &st);
+	if (ret) {
+		tcmu_dev_err(dev, "glfs_lstat failed: %m\n");
+		return ret;
+	}
+
+	if (st.st_size != new_size) {
+		tcmu_dev_err(dev, "Mismatched sizes. glfs image size %lld. Requested new size %" PRIu64 ".\n",
```
Comment (dillaman) [inline glfs.c]:
```
In file included from /home/XXX/tcmu-runner/tcmu-runner.h:33:0,
                 from /home/XXX/tcmu-runner/glfs.c:32:
/home/XXX/tcmu-runner/glfs.c: In function ‘tcmu_glfs_get_image_size’:
/home/XXX/tcmu-runner/glfs.c:652:84: error: expected ‘)’ before ‘PRIu64’
   tcmu_dev_err(dev, "Mismatched sizes. glfs image size %lld. Requested new size %" PRIu64 ".\n",
                                                                                    ^
/home/XXX/tcmu-runner/libtcmu_log.h:71:76: note: in definition of macro ‘tcmu_dev_err’
 #define tcmu_dev_err(dev, ...)  {tcmu_err_message(dev, __func__, __LINE__, __VA_ARGS__);}
```

Comment (dillaman) [reply glfs.c]: ... missing ```#include <inttypes.h>```

Comment (mikechristie) [reply glfs.c]: Thanks Jason.

I fixed this for you @pkalever.

**Comment by pkalever** [comment]
Thanks @dillaman @mikechristie 

Mike,

I hope resize feature is not missing anything (which I'm unaware about ) and it is complete, please correct me if I'm wrong. I just got a bit confused at https://github.com/open-iscsi/tcmu-runner/pull/309 after noting

> Notes:
> 
> This patchset does not coordinate with the kernel to clear the UA queue when there have been resets in the kernel.

> I only added resize support. I did not do the other ones in PR 189 yet.

I hope in the above line you are pointing to TCMULIB_CFG_DEV_CFGSTR, which still needs to be addressed, is that right ?

**Pull request merged**